frameworks/kafka/src/main/java/com/mesosphere/sdk/kafka/api/TopicController.java

```diff
@@ -0,0 +1,171 @@
+        try {
+            JSONObject obj = cmdExecutor.underReplicatedPartitions();
+            return Response.ok(obj.toString(), MediaType.APPLICATION_JSON).build();
+        } catch (Exception ex) {
+            log.error("Failed to fetch topics with exception: " + ex);
+            return Response.serverError().build();
+        }
+    }
+
+    @PUT
+    @Path("/{name}")
+    public Response operationOnTopic(
+            @PathParam("name") String name,
+            @QueryParam("operation") String operation,
+            @QueryParam("key") String key,
+            @QueryParam("value") String value,
+            @QueryParam("partitions") String partitions,
+            @QueryParam("messages") String messages) {
+
+        try {
+            JSONObject result = null;
+            List<String> cmds = null;
+
+            if (operation == null) {
+                result = new JSONObject();
+                result.put("Error", "Must designate an 'operation'.  " +
+                        "Possibles operations are [producer-test, delete, partitions, config, deleteConfig].");
+            } else {
+                switch (operation) {
+                    case "producer-test":
+                        int messageCount = Integer.parseInt(messages);
+                        result = cmdExecutor.producerTest(name, messageCount);
+                        break;
+                    case "partitions":
+                        cmds = Arrays.asList("--partitions", partitions);
+                        result = cmdExecutor.alterTopic(name, cmds);
+                        break;
+                    default:
+                        result = new JSONObject();
+                        result.put("Error", "Unrecognized operation: " + operation);
+                        break;
+                }
+            }
+
+            return Response.ok(result.toString(), MediaType.APPLICATION_JSON).build();
+
+        } catch (Exception ex) {
+            log.error("Failed to perform operation: " + operation + " on Topic: " + name + " with exception: " + ex);
+            return Response.serverError().build();
+        }
+    }
+
+    @DELETE
+    @Path("/{name}")
+    public Response deleteTopic(
+            @PathParam("name") String name) {
+
+        try {
+            JSONObject result = cmdExecutor.deleteTopic(name);
+            String message = result.getString("message");
+            if (message.contains("This will have no impact if delete.topic.enable is not set to true")) {
+                return Response.accepted().entity(result.toString()).type(MediaType.APPLICATION_JSON).build();
+            } else {
+                return Response.ok(result.toString(), MediaType.APPLICATION_JSON).build();
+            }
+
+        } catch (Exception ex) {
+            log.error("Failed to delete Topic: " + name + " with exception: " + ex);
+            return Response.serverError().build();
+        }
+    }
+
+    @GET
+    @Path("/{name}/offsets")
```

Comment:
`/{name}...` could mean that `{name}` is a topic or an operation depending on context, these really need to be in different subpaths to avoid that ambiguity.